cosmo_tester/test_suites/test_snapshots/snapshots_migration_from_321_and_331_to_34_test.py

```diff
@@ -0,0 +1,276 @@
+import json
+import os
+import subprocess
+import tempfile
+
+import pkg_resources
+
+import cosmo_tester
+from cosmo_tester.framework.git_helper import clone
+from cosmo_tester.framework.testenv import (
+    TestCase, initialize_without_bootstrap, clear_environment)
+from cosmo_tester.framework.util import YamlPatcher
+
+MANAGER_KEY_FILE_NAME = 'manager_kp.pem'
+AGENTS_KEY_FILE_NAME = 'agents_kp.pem'
+BLUEPRINTS_REPO = 'https://github.com/cloudify-cosmo/' \
+                  'cloudify-manager-blueprints.git'
+CLI_REPO = 'https://github.com/cloudify-cosmo/cloudify-cli.git'
+HELLO_WORLD_REPO = 'https://github.com/cloudify-cosmo/' \
+                   'cloudify-hello-world-example'
+SNAPSHOT_TOOL_REPO = 'https://github.com/cloudify-cosmo/' \
+                     'cloudify-3.2.1-snapshots-tool'
+
+BRANCH_NAMES = {
+    '3.3.1': '3.3.1-build',
+    '3.2.1': '3.2.1-build',
+    '3.4': '3.4rc1'
+}
+
+
+def setUp():
+    initialize_without_bootstrap()
+
+
+def tearDown():
+    clear_environment()
+
+
+class HelloWorldSnapshotMigrationFrom_3_3_To_3_4_Test(TestCase):
+    def setUp(self):
+        super(HelloWorldSnapshotMigrationFrom_3_3_To_3_4_Test, self).setUp()
+
+    def _bootstrap_manager(self, version):
+        manager_dir = self._get_manager_dir(version)
+        venv_dir = self._get_venv_dir(version)
+        activate_path = self._get_activate_path(version)
+        cli_repo_dir = clone(CLI_REPO, manager_dir, BRANCH_NAMES[version])
+        cli_requirements_path = os.path.join(cli_repo_dir,
+                                             'dev-requirements.txt')
+        blueprint_path = self._prepare_manager_blueprint(version)
+        inputs_path = self._prepare_manager_inputs(
+            os.path.dirname(blueprint_path), version)
+
+        arguments = manager_dir, venv_dir, activate_path, \
```

Comment:
I think all script arguments should be passed as environment variables instead of positional arguments.
It will make the tests easier to read and easier to extend in the future